network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/failure/FailureRecoveryHandler.kt

```diff
@@ -79,7 +90,7 @@ class FailureRecoveryHandler @Inject constructor(
         var currentDelay = initialDelay
         repeat(times) {
             try {
-                if (!recoveryCompleted) {
+                if (failureRecoveryInProgress) {
                     block()
                 } else {
                     return@incrementalPeriodicChecks
```

Comment:
Should we pixel this to see how many instances do not recover?

Reply:
I was contemplating about that actually. We do have `tunnel_failure` and `tunnel_failure_recovered` but emitted from `WireguardHandshakeMonitor` already. That will already tell us how many tunnels failed and how many recovered. The only thing is that includes the ones that doesn’t trigger failure recovery (since we have a high threshold there).